src/actions/vdom/add-attribute.js

```diff
@@ -0,0 +1,16 @@
+var toCamelCase = require('../../utils/to-camel-case');
+
+module.exports = function addAttribute(node, key, val) {
+
+  node = (node.vNode || node);
+  node.properties.dataset = (node.properties.dataset || {});
+
+  if (/data/.test(key)) {
```

Comment:
Yeah, (off the top of my head) theres a difference between `virtual-hyperscript` and `vtree`.  If my memory servers me correctly `virtual-hyperscript` chose to drop the `dataset` object where as `vtree` needs to be updated.